File: 0075-sort-colors/0075-sort-colors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        a, b, c = 0, 0, len(nums)-1
        
        while b <= c:
            if nums[b] == 0:
                nums[a], nums[b] = nums[b], nums[a]
                a += 1
            
            elif nums[b] == 2:
                nums[b], nums[c] = nums[c], nums[b]
                b -= 1
                c -= 1
            
            b += 1
        
        return nums

File: 0075-sort-colors/0075-sort-colors.ts
/**
 Do not return anything, modify nums in-place instead.
 */
function sortColors(nums: number[]): void {
    let a = 0, b = 0, c = nums.length-1
    while (b <= c) {
        if (nums[b] === 0) {
            [nums[a], nums[b]] = [nums[b], nums[a]]
            a++
        } else if (nums[b] === 2) {
            [nums[b], nums[c]] = [nums[c], nums[b]]
            b--
            c--
        }
        b++
    }
};